csclient/csclient_test.go

```diff
@@ -1700,6 +1700,28 @@ func (s *suite) TestUploadResource(c *gc.C) {
 	}
 }
 
+func (s *suite) TestUploadEmptyResource(c *gc.C) {
+	ch := charmtesting.NewCharmMeta(&charm.Meta{
+		Resources: map[string]resource.Meta{
+			"resname": {
+				Name: "resname",
+				Path: "foo.zip",
+			},
+		},
+	})
+	url, err := s.client.UploadCharm(charm.MustParseURL("cs:~who/trusty/mysql"), ch)
+	c.Assert(err, gc.IsNil)
+
+	// Upload the resource.
+	data := ""
+	rev, err := s.client.UploadResource(url, "resname", "data.zip", strings.NewReader(data), int64(len(data)), nil)
```

Comment:
Can't you just pass nil and 0 here?

Reply:
probably, but i don't think that would be quite as obvious... also, it's important not to pass nil for the reader, because other clients won't.